File: scripting-langs/ruby/parse_csv.rb
require 'csv'

def print_header(filename)
  headers = CSV.read(filename, headers: true).headers
  puts "Header: #{headers}"
end

def print_contents(filename)
  CSV.foreach(filename, headers: true) do |row|
    puts row.fields.inspect
  end
end

print_header("iris.csv")
print_contents("iris.csv")

File: scripting-langs/python/parse_csv.py
import csv

def print_header(filename):
    with open(filename, newline='') as file:
        reader = csv.reader(file)
        header = next(reader)
        print("Header:", header)

def print_contents(filename):
    with open(filename, newline='') as file:
        reader = csv.reader(file)
        next(reader)  # skip header
        for row in reader:
            print(row)

print_header("iris.csv")
print_contents("iris.csv")


